### PR TITLE
feat(eg): add new resource endpoint

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/eg' TESTARGS='-run TestAccEndpoint_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eg -v -run TestAccEndpoint_basic -timeout 360m -parallel 4
=== RUN   TestAccEndpoint_basic
=== PAUSE TestAccEndpoint_basic
=== CONT  TestAccEndpoint_basic
--- PASS: TestAccEndpoint_basic (62.39s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eg        62.441s
```

### CODE REVIEW

File: huaweicloud/config/config.go [hunk 1]
```diff
@@ -861,6 +861,10 @@ func (c *Config) ServiceStageV2Client(region string) (*golangsdk.ServiceClient,
 	return c.NewServiceClient("servicestagev2", region)
 }
 
+func (c *Config) EGClient(region string) (*golangsdk.ServiceClient, error) {
```
Comment (Lance52259) [inline huaweicloud/config/config.go]: This method is never used

File: huaweicloud/services/acceptance/eg/resource_huaweicloud_eg_endpoint_test.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+func testEndpoint_basic_update(name string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_eg_endpoint" "test" {
+  name        = "%s"
+  vpc_id      = huaweicloud_vpc.test.id
+  subnet_id   = huaweicloud_vpc_subnet.test.id
+  description = "created by terraform update"
+}
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/eg/resource_huaweicloud_eg_endpoint_test.go]:
```suggestion
resource "huaweicloud_eg_endpoint" "test" {
  name      = "%s"
  vpc_id    = huaweicloud_vpc.test.id
  subnet_id = huaweicloud_vpc_subnet.test.id
}
```
both empty value and update logic we can check

File: docs/resources/eg_endpoint.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
+---
+subcategory: "EventGrid (EG)"
+---
+
+# huaweicloud_eg_endpoint
+
+Manages a EG endpoint resource within HuaweiCloud.
```
Comment (Lance52259) [inline docs/resources/eg_endpoint.md]:
```suggestion
Manages an EG endpoint resource within HuaweiCloud.
```

File: docs/resources/eg_endpoint.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+---
+subcategory: "EventGrid (EG)"
```
Comment (Lance52259) [inline docs/resources/eg_endpoint.md:2]:
```suggestion
subcategory: "Event Grid (EG)"
```

File: huaweicloud/config/config.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply huaweicloud/config/config.go]: fixed

File: huaweicloud/services/acceptance/eg/resource_huaweicloud_eg_endpoint_test.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/eg/resource_huaweicloud_eg_endpoint_test.go]: sure

File: docs/resources/eg_endpoint.md (hunk 3 above)
Comment (Jason-Zhang9309) [reply docs/resources/eg_endpoint.md]: fixed

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/eg' TESTARGS='-run TestAccEndpoint_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eg -v -run TestAccEndpoint_basic -timeout 360m -parallel 4
=== RUN   TestAccEndpoint_basic
=== PAUSE TestAccEndpoint_basic
=== CONT  TestAccEndpoint_basic
--- PASS: TestAccEndpoint_basic (64.05s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eg        64.103s
```

File: huaweicloud/services/eg/resource_huaweicloud_eg_endpoint.go [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+		d.Set("status", utils.PathSearch("status", getEndpointRespBody, nil)),
+		d.Set("created_at", utils.PathSearch("created_time", getEndpointRespBody, nil)),
+		d.Set("updated_at", utils.PathSearch("updated_time", getEndpointRespBody, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func BuildGetEndpointQueryParams() string {
+	res := "?limit=100"
```
Comment (ShiChangkuo) [inline huaweicloud/services/eg/resource_huaweicloud_eg_endpoint.go:212]: how about query endpoints with **name**?

```go
if name != "" {
    res += fmt.Sprintf("&name=%s", name)
}
```

Comment (Jason-Zhang9309) [reply huaweicloud/services/eg/resource_huaweicloud_eg_endpoint.go:212]: sure

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/eg' TESTARGS='-run TestAccEndpoint_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eg -v -run TestAccEndpoint_basic -timeout 360m -parallel 4
=== RUN   TestAccEndpoint_basic
=== PAUSE TestAccEndpoint_basic
=== CONT  TestAccEndpoint_basic
--- PASS: TestAccEndpoint_basic (62.66s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eg        62.719s
```

**Pull request merged**